### ISSUE TITLE
redirecting to wss instead of ws on localhost v1.3

### ISSUE THREAD

**Issue by sairum123** [beyondcode/laravel-websockets#378]
I am using laravel version 5.7 and beyondcode\laravel-websockets package v1.3 the problem I am facing right now is when I am running it on localhost I am getting this error
 
```WebSocket connection to 'wss://localhost/app/somekey?protocol=7&client=js&version=6.0.2&flash=false' failed: Error in connection establishment: net::ERR_CERT_AUTHORITY_INVALID```

I have changed encrypted to false in bootsrap.js still it connects to wss instead of ws

bootstrap.js

```
import Echo from 'laravel-echo'

window.Pusher = require('pusher-js');

window.Echo = new Echo({
    broadcaster: 'pusher',
    key: process.env.MIX_PUSHER_APP_KEY,
    cluster: process.env.MIX_PUSHER_APP_CLUSTER,
    encrypted: false,
    wsHost: window.location.hostname,
    wsPort: 6001,
    disableStats: true
});
```
broadcasting.php config
```
'connections' => [

        'pusher' => [
            'driver' => 'pusher',
            'key' => env('PUSHER_APP_KEY'),
            'secret' => env('PUSHER_APP_SECRET'),
            'app_id' => env('PUSHER_APP_ID'),
            'options' => [
                'cluster' => env('PUSHER_APP_CLUSTER'),
                'encrypted' => false,
                'host' => '127.0.0.1',
                'port' => 6001,
                'scheme' => 'http'
            ],
        ],
```

**Comment by intergalactisch** [comment]
I also have this same error. Is this a Chrome or a Laravel thing?

**Comment by sairum123** [comment]
its a laravel thing

**Comment by pietrantonio91** [comment]
Same issue, following.

**Comment by pietrantonio91** [comment]
I solved by downgrading pusher-js to 4.3

**Comment by KuenzelIT** [comment]
Maybe the config options for pusher-js 6 changed, which are not reflected in Echo yet? I may look into it later.

**Comment by mralston** [comment]
TL;DR - set `forceTLS: false` when you bootstrap Echo:

```js
window.Echo = new Echo({
    broadcaster: 'pusher',
    key: process.env.MIX_PUSHER_APP_KEY,
    wsHost: window.location.hostname,
    wsPort: 6001,
    disableStats: true,
    forceTLS: false // Critical if you want to use a non-secure WebSocket connection
});
```

Same problem. I did a little experimenting.

I found that the `useTLS` was set to `true` when the code inside `pusher-js` attempts establish the web socket connection. I was unable to change `useTLS` to `false` when bootstrapping Echo, `useTLS` is always `true` inside `pusher-js`.

If I modify the `pusher-js` code as follows, it connects:

```js
TransportConnection.prototype.connect = function () {
        var _this = this;
        if (this.socket || this.state !== 'initialized') {
            return false;
        }

        // Add this line to override force useTLS=false
        this.options.useTLS = false;

        var url = this.hooks.urls.getInitial(this.key, this.options);
        try {
            this.socket = this.hooks.getSocket(url, this.options);
        }
        ...
```

Obviously, this isn't a workable solution, this modification will be overwritten the next time npm compiles the JS.

I don't know whether this is new, this is the first time I've attempted to use web sockets or Pusher, but I suspect this may be the culprit, in `pusher-js`:

```js
function shouldUseTLS(opts) {
    if (runtime.getProtocol() === 'https:') {
        return true;
    }
    else if (opts.forceTLS === false) {
        return false;
    }
    return true;
}
```

See how it returns `true` by default? The result of this function is used to populate `useTLS`.

From that function, I deduced that setting the option `forceTLS` to `false` would solve the issue (and it does).

I'm not sure yet why `pusher-js` is overriding the `useTLS` option, presumably Pusher are phasing out all non-secure connections to their platform.

**Comment by akyagmur** [comment]
@mralston Thanks! forceTLS option solved my issue.

**Comment by hallowcard13** [comment]
> Same problem. I did a little experimenting.
> 
> TL;DR - set `forceTLS: false` when you bootstrap Echo:
> 
> ```js
> window.Echo = new Echo({
>     broadcaster: 'pusher',
>     key: process.env.MIX_PUSHER_APP_KEY,
>     wsHost: window.location.hostname,
>     wsPort: 6001,
>     disableStats: true,
>     forceTLS: false // Critical if you want to use a non-secure WebSocket connection
> });
> ```
> 
> Explanation:
> 
> I found that the `useTLS` was set to `true` when the code inside `pusher-js` attempts establish the web socket connection. I was unable to change `useTLS` to `false` when bootstrapping Echo, `useTLS` is always `true` inside `pusher-js`.
> 
> If I modify the `pusher-js` code as follows, it connects:
> 
> ```js
> TransportConnection.prototype.connect = function () {
>         var _this = this;
>         if (this.socket || this.state !== 'initialized') {
>             return false;
>         }
> 
>         // Add this line to override force useTLS=false
>         this.options.useTLS = false;
> 
>         var url = this.hooks.urls.getInitial(this.key, this.options);
>         try {
>             this.socket = this.hooks.getSocket(url, this.options);
>         }
>         ...
> ```
> 
> Obviously, this isn't a workable solution, this modification will be overwritten the next time npm compiles the JS.
> 
> This may be the culprit, in `pusher-js`:
> 
> ```js
> function shouldUseTLS(opts) {
>     if (runtime.getProtocol() === 'https:') {
>         return true;
>     }
>     else if (opts.forceTLS === false) {
>         return false;
>     }
>     return true;
> }
> ```
> 
> See how it returns `true` by default? The result of this function is used to populate the `useTLS` option.
> 
> From that function, I deduced that setting the option `forceTLS` to `false` would solve the issue (and it does).

unfornately this not working now.

**Comment by robgwin** [comment]
```
    if (runtime.getProtocol() === 'https:') {
        return true;
    }
```

I believe that ☝️ means pusher-js will always force TLS if your main laravel app is running on https. You can use forceTLS to run secure sockets in an unsecure app, but there's no forceNotTLS option to do the reverse.

Which honestly probably makes sense as an overall policy -- pretty bad idea to have an insecure data channel inside an app that "looks" secure. I just ended up here because I'm testing this out in my dev environment and wanted to make sure everything was working before dealing with SSL certs. Oh well.
